File: boj/13877/soln13877.go
/*
// 백준
// No. 13877
// Go 1.24.8
// by "nno0obb"
*/

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

var reader = bufio.NewReader(os.Stdin)
var writer = bufio.NewWriter(os.Stdout)

func main() {
	defer writer.Flush()

	// Input
	var T int
	fmt.Fscanln(reader, &T)
	for range T {
		var K, S string
		fmt.Fscanln(reader, &K, &S)

		// Logic
		_8, err := strconv.ParseInt(S, 8, 64)
		if err != nil {
			_8 = 0
		}
		_10, _ := strconv.ParseInt(S, 10, 64)
		_16, _ := strconv.ParseInt(S, 16, 64)

		// Output
		fmt.Fprintln(writer, K, _8, _10, _16)
	}
}

File: boj/13877/soln13877.py
"""
# 백준
# No. 13877
# Python 3.11.9
# by "nno0obb"
"""

import sys


def main():
    # Input
    T = int(input())
    for _ in range(T):
        K, S = sys.stdin.readline().split()

        # Logic
        try:
            _8 = int(S, 8)
        except ValueError:
            _8 = "0"
        _10 = int(S)
        _16 = int(S, 16)

        # Output
        print(K, _8, _10, _16)


if __name__ == "__main__":
    main()
